app/assets/javascripts/specifics.js.coffee

```diff
@@ -564,7 +590,23 @@ class Row
   toHashKey: ->
     @flattenToIds().join(",") + ",#{@specificLeftMost}" + ",#{@nonSpecificLeftMost?.id}"
 
-  
+  # If the row references a leftmost, either specific or not, return the event(s)
+  # (because a UNION can place multiple events in the specific leftMost, this can be > 1)
+  leftMostEvents: ->
+    if @nonSpecificLeftMost?
+      return [@nonSpecificLeftMost]
+    if @specificLeftMost? && _.isString(@specificLeftMost)
+      specificIndex = hqmf.SpecificsManager.getColumnIndex(@specificLeftMost)
+      return [@values[specificIndex]] if @values[specificIndex]?
+    if @specificLeftMost? && _.isObject(@specificLeftMost)
+      events = []
+      for id, occurrences of @specificLeftMost
+        for occurrence in occurrences
+          specificIndex = hqmf.SpecificsManager.getColumnIndex(occurrence)
+          events.push(@values[specificIndex]) if @values[specificIndex]?
+      return events
+    return []
+
```

Comment:
This looks like a repeat (and slightly different implementation) of the code in the `for` loop in `extractEventsForLeftMost`. Consider using this in the `for` loop instead. Also, double check implementation differences (I'm pretty sure it's the same result, but worth taking a look).